core/primitives/src/shard_layout.rs

```diff
@@ -225,25 +221,18 @@ impl ShardLayout {
 /// Maps an account to the shard that it belongs to given a shard_layout
 /// For V0, maps according to hash of account id
 /// For V1, accounts are divided to ranges, each range of account is mapped to a shard.
-/// There are also some fixed shards, each of which is mapped to an account and all sub-accounts.
-///     For example, for ShardLayoutV1{ fixed_shards: ["aurora"], boundary_accounts: ["near"]}
-///     Account "aurora" and all its sub-accounts will be mapped to shard_id 0.
-///     For the rest of accounts, accounts <= "near" will be mapped to shard_id 1 and
-///     accounts > "near" will be mapped shard_id 2.
 pub fn account_id_to_shard_id(account_id: &AccountId, shard_layout: &ShardLayout) -> ShardId {
     match shard_layout {
         ShardLayout::V0(ShardLayoutV0 { num_shards, .. }) => {
             let hash = CryptoHash::hash_bytes(account_id.as_ref().as_bytes());
             let (bytes, _) = stdx::split_array::<32, 8, 24>(hash.as_bytes());
             u64::from_le_bytes(*bytes) % num_shards
         }
-        ShardLayout::V1(ShardLayoutV1 { fixed_shards, boundary_accounts, .. }) => {
-            for (shard_id, fixed_account) in fixed_shards.iter().enumerate() {
-                if is_top_level_account(fixed_account, account_id) {
-                    return shard_id as ShardId;
-                }
-            }
-            let mut shard_id = fixed_shards.len() as ShardId;
+        ShardLayout::V1(ShardLayoutV1 { boundary_accounts, .. }) => {
+            // Note: As we scale up the number of shards we can consider
+            // changing this method to do a binary search rather than linear
+            // scan. For the time being, with only 4 shards, this is perfectly fine.
+            let mut shard_id = 0 as ShardId;
```

Comment:
super nit: this looks a bit weird, `let mut shard_id: ShardId = 0` seems more canonical to me